packages/sitecore-jss-nextjs/src/components/Placeholder.tsx

```diff
@@ -11,7 +11,7 @@ export const Placeholder = (props: PlaceholderComponentProps) => {
   return (
     <ReactPlaceholder
       {...props}
-      modifyComponentProps={(initialProps) => {
+      modifyComponentProps={(initialProps: { rendering: { uid: string | number } }) => {
```

Comment:
I left this in order for us to make a decision regarding the _SitecoreContext_ optional or required. Accordingly porotypes would change.

Reply:
I think the introduction of `sitecoreContext` to the `PlaceholderComponentProps` might have influenced TypeScript's type inference or exposed previously overlooked implicit 'any' usage. I don't see anything else causing this as `sitecoreContext` is optional prop.

Reply:
Previously, withComponentFactory HOC would return a placeholder with the same props type. withSitecoreContext returns Placeholder with "WithSitecoreContextHocProps" prop type.
I think this is the culprit. typescript considers this as "any" type because it cannot find the "modifyComponentProps" field in WithSitecoreContextHocProps.